Snowflake receiver: conversion error for 'bytesInserted' trying to convert float64 to int64

### Component(s)

_No response_

### What happened?

In snowflake receiver, for one of the snowflake pipe metric 

we have the data type defined as int https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/snowflakereceiver/model.go#L89

but in reality when parsing the SQL query https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/snowflakereceiver/client.go#L24 and scanning the related column value sum(bytes_inserted) the receiver gets float value but expects int and hence the error.


```
May 19 13:05:16 MEIPLAP01089 otelcol[1304954]: 2025-05-19T13:05:16.188+0200        error        scraperhelper@v0.125.0/obs_metrics.go:61        Error scraping metrics        {"otelcol.component.id": "snowflake/azure_ci_prod_xs", "otelcol.component.kind": "receiver", "otelcol.signal": "metrics", "error": "sql: Scan error on column index 2, name \"SUM(BYTES_INSERTED)\": converting driver.Value type float64 (\"4.8225792e+08\") to a int64: invalid syntax"}
We might have to make additional checks for any other such metrics where expect int but actually get float.
```

Another underlying issue:

Just the presence of snowflake receiver will throw this error because it seems the receiver issues query and parses all metrics whether or not that metric is enabled. It is only after the scanning and parsing of the metrics, metric is checked whether it is enabled or not and if not enabled it is not added to the list of datapoints to be sent.

Steps to reproduce:
1. snowflake receiver


Expected: 
No scan error when trying to parse snowflake metrics

Actual:
```
"error": "sql: Scan error on column index 2, name \"SUM(BYTES_INSERTED)\": converting driver.Value type float64 (\"4.8225792e+08\") to a int64: invalid syntax"
```

### Collector version

0.126.0

### Environment information

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")


### OpenTelemetry Collector configuration

```yaml

```

### Log output

```shell

```

### Additional context

_No response_

--- a/.chloggen/40418-conversionerror-bugfix.yaml
+++ b/.chloggen/40418-conversionerror-bugfix.yaml
@@ -1,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: snowflakereceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: fixes bug regarding a bad type conversion and adds guard rails to prevent future similar bugs
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [40418]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: []

--- a/.chloggen/40418-conversionerror-bugfix.yaml
+++ b/.chloggen/40418-conversionerror-bugfix.yaml
@@ -7,7 +7,7 @@ change_type: bug_fix
 component: snowflakereceiver
 
 # A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
-note: fixes bug regarding a bad type conversion and adds guard rails to prevent future similar bugs
+note: adds a pre-scrape check to see if metrics are enabled.
 
 # Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
 issues: [40418]

--- a/receiver/snowflakereceiver/scraper.go
+++ b/receiver/snowflakereceiver/scraper.go
@@ -107,6 +107,10 @@ func (s *snowflakeMetricsScraper) scrape(ctx context.Context) (pmetric.Metrics, 
 }
 
 func (s *snowflakeMetricsScraper) scrapeBillingMetrics(ctx context.Context, t pcommon.Timestamp, errs chan<- error) {
+	if !s.conf.Metrics.SnowflakeBillingCloudServiceTotal.Enabled && !s.conf.Metrics.SnowflakeBillingTotalCreditTotal.Enabled && !s.conf.Metrics.SnowflakeBillingVirtualWarehouseTotal.Enabled {
+		return
+	}
+
 	billingMetrics, err := s.client.FetchBillingMetrics(ctx)
 	if err != nil {
 		errs <- err
@@ -121,6 +125,10 @@ func (s *snowflakeMetricsScraper) scrapeBillingMetrics(ctx context.Context, t pc
 }
 
 func (s *snowflakeMetricsScraper) scrapeWarehouseBillingMetrics(ctx context.Context, t pcommon.Timestamp, errs chan<- error) {
+	if !s.conf.Metrics.SnowflakeBillingWarehouseTotalCreditTotal.Enabled && !s.conf.Metrics.SnowflakeBillingWarehouseCloudServiceTotal.Enabled && !s.conf.Metrics.SnowflakeBillingWarehouseVirtualWarehouseTotal.Enabled {
+		return
+	}
+
 	warehouseBillingMetrics, err := s.client.FetchWarehouseBillingMetrics(ctx)
 	if err != nil {
 		errs <- err
@@ -135,6 +143,10 @@ func (s *snowflakeMetricsScraper) scrapeWarehouseBillingMetrics(ctx context.Cont
 }
 
 func (s *snowflakeMetricsScraper) scrapeLoginMetrics(ctx context.Context, t pcommon.Timestamp, errs chan<- error) {
+	if !s.conf.Metrics.SnowflakeLoginsTotal.Enabled {
+		return
+	}
+
 	loginMetrics, err := s.client.FetchLoginMetrics(ctx)
 	if err != nil {
 		errs <- err
@@ -147,6 +159,9 @@ func (s *snowflakeMetricsScraper) scrapeLoginMetrics(ctx context.Context, t pcom
 }
 
 func (s *snowflakeMetricsScraper) scrapeHighLevelQueryMetrics(ctx context.Context, t pcommon.Timestamp, errs chan<- error) {
+	if !s.conf.Metrics.SnowflakeQueryExecuted.Enabled && !s.conf.Metrics.SnowflakeQueryBlocked.Enabled && !s.conf.Metrics.SnowflakeQueryQueuedOverload.Enabled && !s.conf.Metrics.SnowflakeQueryQueuedProvision.Enabled {
+		return
+	}
 	highLevelQueryMetrics, err := s.client.FetchHighLevelQueryMetrics(ctx)
 	if err != nil {
 		errs <- err
@@ -162,6 +177,18 @@ func (s *snowflakeMetricsScraper) scrapeHighLevelQueryMetrics(ctx context.Contex
 }
 
 func (s *snowflakeMetricsScraper) scrapeDBMetrics(ctx context.Context, t pcommon.Timestamp, errs chan<- error) {
+	if !s.conf.Metrics.SnowflakeDatabaseQueryCount.Enabled && !s.conf.Metrics.SnowflakeDatabaseBytesScannedAvg.Enabled &&
+		!s.conf.Metrics.SnowflakeQueryBytesDeletedAvg.Enabled && !s.conf.Metrics.SnowflakeQueryBytesSpilledLocalAvg.Enabled &&
+		!s.conf.Metrics.SnowflakeQueryBytesSpilledRemoteAvg.Enabled && !s.conf.Metrics.SnowflakeQueryBytesWrittenAvg.Enabled &&
+		!s.conf.Metrics.SnowflakeQueryCompilationTimeAvg.Enabled && !s.conf.Metrics.SnowflakeQueryDataScannedCacheAvg.Enabled &&
+		!s.conf.Metrics.SnowflakeQueryExecutionTimeAvg.Enabled && !s.conf.Metrics.SnowflakeQueryPartitionsScannedAvg.Enabled &&
+		!s.conf.Metrics.SnowflakeQueuedOverloadTimeAvg.Enabled && !s.conf.Metrics.SnowflakeQueuedProvisioningTimeAvg.Enabled &&
+		!s.conf.Metrics.SnowflakeQueuedRepairTimeAvg.Enabled && !s.conf.Metrics.SnowflakeRowsInsertedAvg.Enabled &&
+		!s.conf.Metrics.SnowflakeRowsDeletedAvg.Enabled && !s.conf.Metrics.SnowflakeRowsProducedAvg.Enabled &&
+		!s.conf.Metrics.SnowflakeRowsUnloadedAvg.Enabled && !s.conf.Metrics.SnowflakeRowsUpdatedAvg.Enabled &&
+		!s.conf.Metrics.SnowflakeTotalElapsedTimeAvg.Enabled {
+		return
+	}
 	DBMetrics, err := s.client.FetchDbMetrics(ctx)
 	if err != nil {
 		errs <- err
@@ -192,6 +219,10 @@ func (s *snowflakeMetricsScraper) scrapeDBMetrics(ctx context.Context, t pcommon
 }
 
 func (s *snowflakeMetricsScraper) scrapeSessionMetrics(ctx context.Context, t pcommon.Timestamp, errs chan<- error) {
+	if !s.conf.Metrics.SnowflakeSessionIDCount.Enabled {
+		return
+	}
+
 	sessionMetrics, err := s.client.FetchSessionMetrics(ctx)
 	if err != nil {
 		errs <- err
@@ -204,6 +235,10 @@ func (s *snowflakeMetricsScraper) scrapeSessionMetrics(ctx context.Context, t pc
 }
 
 func (s *snowflakeMetricsScraper) scrapeSnowpipeMetrics(ctx context.Context, t pcommon.Timestamp, errs chan<- error) {
+	if !s.conf.Metrics.SnowflakePipeCreditsUsedTotal.Enabled {
+		return
+	}
+
 	snowpipeMetrics, err := s.client.FetchSnowpipeMetrics(ctx)
 	if err != nil {
 		errs <- err
@@ -216,6 +251,10 @@ func (s *snowflakeMetricsScraper) scrapeSnowpipeMetrics(ctx context.Context, t p
 }
 
 func (s *snowflakeMetricsScraper) scrapeStorageMetrics(ctx context.Context, t pcommon.Timestamp, errs chan<- error) {
+	if !s.conf.Metrics.SnowflakeStorageStorageBytesTotal.Enabled && !s.conf.Metrics.SnowflakeStorageStageBytesTotal.Enabled && !s.conf.Metrics.SnowflakeStorageFailsafeBytesTotal.Enabled {
+		return
+	}
+
 	storageMetrics, err := s.client.FetchStorageMetrics(ctx)
 	if err != nil {
 		errs <- err

